pointerevents/pointerlock/pointerevent_movementxy-manual.html

```diff
@@ -0,0 +1,103 @@
+<!doctype html>
+<html>
+    <head>
+        <title>Pointer Events properties tests</title>
+        <meta name="viewport" content="width=device-width">
+        <link rel="stylesheet" type="text/css" href="../pointerevent_styles.css">
+        <script src="/resources/testharness.js"></script>
+        <script src="/resources/testharnessreport.js"></script>
+        <!-- Additional helper script for common checks across event types -->
+        <script type="text/javascript" src="../pointerevent_support.js"></script>
+        <style>
+          body {
+            touch-action: none;
+            user-select: none;
+          }
+          #box1 {
+            top: 330px;
+            left: 150px;
+            background: black;
+          }
+          #box2 {
+            top: 370px;
+            left: 350px;
+            background: red;
+          }
+          #square2 {
+            visibility: block;
+          }
+        </style>
+        <script>
+            var expectedPointerId = NaN;
+            var startSummation = false;
+            var isFirstMove = true;
+            var lastClientX = 0;
+            var lastClientY = 0;
+
+            function resetTestState() {
+                startSummation = false;
+                isFirstMove = true;
+                lastClientX = 0;
+                lastClientY = 0;
+            }
+
+            function run() {
+                var test_pointerEvent = setup_pointerevent_test("pointerevent attributes", ['mouse', 'touch']);
+
+                [document, document.getElementById('innerFrame').contentDocument].forEach(function(element) {
+                  on_event(element, 'pointermove', function (event) {
+                    if (startSummation) {
+                      if (!isFirstMove) {
+                          test_pointerEvent.step(function() {
+                            assert_equals(event.movementX, event.clientX - lastClientX, "movementX should be the delta between current event's and last event's clientX");
+                            assert_equals(event.movementY, event.clientY - lastClientY, "movementY should be the delta between current event's and last event's clientY");
+                          });
+                      }
+                      isFirstMove = false;
+                      lastClientX = event.clientX;
+                      lastClientY = event.clientY;
+                    }
+                  });
+                });
+                ['pointerenter', 'pointerleave'].forEach(function(eventName) {
+                  on_event(document.getElementById('innerFrame').contentDocument, eventName, function (event) {
+                    isFirstMove = true;
```

Comment:
I see, so this means we're not testing the movementX/movementY values for the event that enters/leaves the iframe.  That's OK but not ideal (misses some interesting test coverage).

The simplest way to get this additional test coverage is to use `screenX/Y` instead of `clientX/Y` (and remove all the `isFirstMove` logic).  That would also better match the spec, since `movementX/Y` is [defined in terms of screen co-ordinates](https://w3c.github.io/pointerlock/#dom-mouseevent-movementx).  Eg. if the browser content area happened to move relative to the physical screen during the test (such as when top controls show or hide in mobile browsers) then the test should still pass when using `screen` co-ordinates but should fail when using `client` (assuming we've implemented according to the spec).

If there's some reason why using client co-ordinates is preferable (eg. perhaps @schieb has a reason why the existing pointerlock tests do that), you could still get this additional test coverage as follows:
 - specify the iframe border size explicitly (with either the 'border=' HTML attribute or a CSS border rule on the iframe element) rather than relying on the browser default (not sure that's standardized).
 - on test start, compute the offset of the iframe within the parent frame by using getBoundingClientRect on innerFrame plus the border size you specified
 - in your pointermove handler, if the event target is inside the iframe, add the above offset to the client co-ordinates

In that way you'd be doing all your math in the ["root frame" coordinate space](https://www.chromium.org/developers/design-documents/blink-coordinate-spaces) (instead of in the client / local-frame co-ordinate space).  And then the math should work out in all cases.